capplets/common/mate-theme-apply.c

```diff
@@ -48,11 +50,26 @@ void
 mate_meta_theme_set (MateThemeMetaInfo *meta_theme_info)
 {
   GSettings *interface_settings;
+  GSettings *interface_gnome_settings;
   GSettings *marco_settings;
   GSettings *mouse_settings;
   GSettings *notification_settings = NULL;
   gchar *old_key;
   gint old_key_int;
+  gboolean wayland;
+
+  /*Find out whether we are running under wayland or x11 */
+
+  if (GDK_IS_X11_DISPLAY ((gdk_display_get_default())))
+  {
+    wayland = FALSE;
+    interface_gnome_settings = NULL;
+  }
+  else
+  {
+    wayland = TRUE;
+    interface_gnome_settings = g_settings_new (INTERFACE_GNOME_SCHEMA);
+  }
```

Comment:
```suggestion

  /*We only want to tough GNOME settings under Wayland */
  if (GDK_IS_X11_DISPLAY (gdk_display_get_default()))
    interface_gnome_settings = NULL;
  else
    interface_gnome_settings = g_settings_new (INTERFACE_GNOME_SCHEMA);
```

Reply:
Latest force-pushed used this simplification, wayland test shows it working